pkg/asset/agent/image/oc.go

```diff
@@ -169,6 +176,62 @@ func (r *release) getImageFromRelease(imageName string) (string, error) {
 
 	return image, nil
 }
+func (r *release) validateReleaseImageArch(architecture string) error {
```

Comment:
IIUC correctly the location of this validation doesn't seem ideal here, for different reasons:
* It could be invoked multiple times (since the `getImageFromRelease` is also part of the `ExtractFile`)
* The validation essentially checks that embedded installer pullspec arch (here retrieved via ClusterImageSet.Spec.ReleaseImage) with the one configured (if defined) in the install-config (here retrieved via InfraEnv.Spec.CpuArchitecture). The goal is thus to ensure that the current installer binary will not be used to generate an agent ISO of a different arch. So this validation, as long as many other validations, should be performed just once in a proper asset (for example in the `OptionalInstallConfig` one)

Reply:
> IIUC correctly the location of this validation doesn't seem ideal here, for different reasons:
> 
> * It could be invoked multiple times (since the `getImageFromRelease` is also part of the `ExtractFile`)
> * The validation essentially checks that embedded installer pullspec arch (here retrieved via ClusterImageSet.Spec.ReleaseImage) with the one configured (if defined) in the install-config (here retrieved via InfraEnv.Spec.CpuArchitecture). The goal is thus to ensure that the current installer binary will not be used to generate an agent ISO of a different arch. 

I think we're on the same page, but just to make sure. The binary of the current installer doesn't matter, the releaseimage architecture of the current installer does.
We can create an agent.aarch64.iso  from an amd64 host as long as the openshift-install binary uses the correct release payload:
https://mirror.openshift.com/pub/openshift-v4/aarch64/clients/ocp/4.13.4/openshift-install-linux-amd64-4.13.4.tar.gz

```
[jeyoung]$ uname -m 
x86_64
[jeyoung]$ ./openshift-install version
./openshift-install 4.13.4
built from commit 90acb3fa2990c35c9beeff4a188fb133fedba432
release image registry.ci.openshift.org/ocp-arm64/release-arm64@sha256:13b14f0514d24d241d40ebacac9f15f93acebc4a7849e4740df49e65e48af424
release architecture arm64
```

> So this validation, as long as many other validations, should be performed just once in a proper asset (for example in the `OptionalInstallConfig` one)

Fair, was hoping to keep in close to the other oc templates. Will reevaluate.

Reply:
> I think we're on the same page, but just to make sure. The binary of the current installer doesn't matter, the releaseimage architecture of the current installer does.

Agree, it's the embedded pullspec that matters

Reply:
@andfasano I've been working on moving the validation to assets. I can get unit/integration tests to pass on my local machine, but I'm worried it will fail in CI b/c `oc` won't be available?

Reply:
Currently the [integration tests in the installer repo](https://github.com/openshift/release/blob/6b9fd78a454b0e9408e1cf783b22fd8bec4009e0/ci-operator/config/openshift/installer/openshift-installer-master.yaml#L688-L692) are run within a specially crafted image `src-oc` that includes the `oc` cli

Reply:
yeah, integration tests are ok. I do see issues when running `hack/go-test.sh` locally because [ docker.io/golang:1.20](https://github.com/openshift/installer/blob/master/hack/go-test.sh#L11 ) doesn't have oc. 

@tvardema  @SaravanaStorageNetwork  Do you all have a suggestion on a go container w/oc that we could use here?

Reply:
Yeap, these tests fail because they use the [ci-operator:latest](https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml#L26) container, which doesn't have `oc`

```                    
ci/prow/aro-unit                    
ci/prow/okd-scos-unit                   
ci/prow/okd-unit                    
ci/prow/unit
```
Basically this PR inadvertently adds a `check to make sure oc is in $PATH` for all agent validation.
Here are some options I see to move fwd:

1. Insert `oc` into `ci-operator:latest`, or find another container that has `oc`. Also find a suitable replacement for [docker.io/golang:1.20](https://github.com/openshift/installer/blob/master/hack/go-test.sh#L11) for local testing. 
2. Only throw a warning if `oc` isn't present during validation (basically skip release arch validation instead of erroring out). 
3. No fix, this kind of validation isn't worth the effort to implement properly. Seems like I'm doing more harm than good here? 
4. Vendor `oc`. 

Thoughts? @zaneb @andfasano @pawanpinjarkar 

FWIW, my preferences in this order are:
1 (only if the needed containers already exist), 2, and lastly 3. 
Not a fan of 4 just for this, but maybe there are other reasons you'd want to vendor that I don't know about?

Thanks again for being patience with this.

Reply:
I vote for 2.
Not opposed to doing 1 as well.
4 unfortunately comes with horrendous dependencies, otherwise we'd do it: https://issues.redhat.com/browse/AGENT-307

Reply:
As long as 2 could be a valid option, there is also the alternative to adopt the `src-oc` image as we already did for the [integration tests](https://github.com/openshift/release/blob/e7613ad8671c48cb9a8c5984fa6406c33da3d1e4/ci-operator/config/openshift/installer/openshift-installer-master.yaml#L689-L692), an [image](https://github.com/openshift/release/blob/e7613ad8671c48cb9a8c5984fa6406c33da3d1e4/ci-operator/config/openshift/installer/openshift-installer-master.yaml#L108-L121) crafted just for including `oc`

Reply:
@andfasano what do you mean by `As long as 2 could be a valid option`? Are you asking for something like `openshift-install agent create image --validate-release-image`

Reply:
I mean, if it's just a test problem then extending the usage of `src-oc` image also to the tests that are not currently using it (but just `src`) should  make them happy. Option 2 seems still a feasible alternative (IIUC, just impacting the test code?)

Reply:
yeah, its breaking ci tests and local test https://github.com/openshift/installer/blob/master/hack/go-test.sh#L11

Reply:
going with option 2, seems unreasonable to force all local testing for the installer to require `oc`?